x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/TelemetryIT.java

```diff
@@ -0,0 +1,269 @@
+                    try {
+                        // test total commands used
+                        final List<Measurement> commandMeasurementsAll = measurements(plugin, PlanningMetricsManager.FEATURE_METRICS_ALL);
+                        assertAllUsages(expectedCommands, commandMeasurementsAll, iteration, success);
+
+                        // test num of queries using a command
+                        final List<Measurement> commandMeasurements = measurements(plugin, PlanningMetricsManager.FEATURE_METRICS);
+                        assertUsageInQuery(expectedCommands, commandMeasurements, iteration, success);
+
+                        // test total functions used
+                        final List<Measurement> functionMeasurementsAll = measurements(plugin, PlanningMetricsManager.FUNCTION_METRICS_ALL);
+                        assertAllUsages(expectedFunctions, functionMeasurementsAll, iteration, success);
+
+                        // test number of queries using a function
+                        final List<Measurement> functionMeasurements = measurements(plugin, PlanningMetricsManager.FUNCTION_METRICS);
+                        assertUsageInQuery(expectedFunctions, functionMeasurements, iteration, success);
+                    } finally {
+                        latch.countDown();
+                    }
+                }));
+                latch.await(30, TimeUnit.SECONDS);
+            }
+        } finally {
+            plugin.resetMeter();
+        }
+
+    }
+
+    private static void assertAllUsages(Map<String, Integer> expected, List<Measurement> metrics, long iteration, Boolean success) {
+        Set<String> found = featureNames(metrics);
+        assertThat(found, is(expected.keySet()));
+        for (Measurement metric : metrics) {
+            assertThat(metric.attributes().get(PlanningMetricsManager.SUCCESS), is(success));
+            String featureName = (String) metric.attributes().get(PlanningMetricsManager.FEATURE_NAME);
+            assertThat(metric.getLong(), is(iteration * expected.get(featureName)));
+        }
+    }
+
+    private static void assertUsageInQuery(Map<String, Integer> expected, List<Measurement> found, long iteration, Boolean success) {
+        Set<String> functionsFound;
+        functionsFound = featureNames(found);
+        assertThat(functionsFound, is(expected.keySet()));
+        for (Measurement measurement : found) {
+            assertThat(measurement.attributes().get(PlanningMetricsManager.SUCCESS), is(success));
+            assertThat(measurement.getLong(), is(iteration));
+        }
+    }
+
+    private static List<Measurement> measurements(TestTelemetryPlugin plugin, String metricKey) {
+        return Measurement.combine(plugin.getLongCounterMeasurement(metricKey));
+    }
+
+    private static Set<String> featureNames(List<Measurement> funcitonMeasurements) {
```

Comment:
Very small typo. `funciton` -> `function`.
Only if you'll have another round of changes in the PR, otherwise it's not worth it.

Reply:
Three in a row, apparently my fingers can't type "function" correctly 😅